### PR TITLE
[Docs] Bundle of errata and small corrections / improvements for torch.linalg docs

### PR DESCRIPTION
**Pull request by lezcano**
This PR bundles a number of errata detected in the linalg docs over the last few weeks.

- Simpler Cholesky deprecation rule
- Remove repeated consecutive words
- Correct cond with rcond in lstsq
- Correct examples of lstsq
- More concise examples
- Use the names of the inputs / outputs in the variables of the examples

### CODE REVIEW

File: aten/src/ATen/native/BatchLinearAlgebra.cpp [hunk 1]
```diff
@@ -1272,7 +1272,7 @@ Tensor cholesky(const Tensor &self, bool upper) {
     "and\n"
     "U = torch.cholesky(A, upper=True)\n",
     "should be replaced with\n",
-    "U = torch.linalg.cholesky(A.transpose(-2, -1).conj()).transpose(-2, -1).conj()"
+    "U = torch.linalg.cholesky(A).transpose(-2, -1).conj()"
```
Comment (IvanYashchuk) [inline aten/src/ATen/native/BatchLinearAlgebra.cpp:1275]: Unfortunately, it's not an equivalent change:
```python
In [1]: import torch

In [2]: A = torch.randn(3, 3, dtype=torch.complex64)

In [3]: A
Out[3]: 
tensor([[-0.3064+0.2171j,  0.2068-1.0801j, -0.0667-0.7361j],
        [-1.3723-1.3536j,  0.7209-0.7423j, -1.6496-0.1502j],
        [-1.2009-0.1911j,  0.6449-0.6975j, -0.1711-0.1355j]])

In [4]: A = A @ A.T.conj()

In [5]: A
Out[5]: 
tensor([[1.8966+0.0000j, 1.2981-0.1335j, 1.3243-0.7546j],
        [1.2981+0.1335j, 7.5298+0.0000j, 3.1919+1.1894j],
        [1.3243+0.7546j, 3.1919-1.1894j, 2.4286+0.0000j]])

In [7]: Atriu = A.triu()

In [8]: torch.linalg.cholesky(Atriu).T.conj() # the result is wrong
Out[8]: 
tensor([[1.3772-0.j, 0.0000-0.j, 0.0000-0.j],
        [0.0000-0.j, 2.7440-0.j, 0.0000-0.j],
        [0.0000-0.j, 0.0000-0.j, 1.5584-0.j]])

In [9]: torch.linalg.cholesky(Atriu.T.conj()).T.conj()
Out[9]: 
tensor([[1.3772-0.0000j, 0.9426-0.0970j, 0.9616-0.5480j],
        [0.0000-0.0000j, 2.5752-0.0000j, 0.8668+0.6262j],
        [0.0000-0.0000j, 0.0000-0.0000j, 0.2451-0.0000j]])

In [10]: torch.cholesky(Atriu, upper=True)
Out[10]: 
tensor([[1.3772+0.0000j, 0.9426-0.0970j, 0.9616-0.5480j],
        [0.0000+0.0000j, 2.5752+0.0000j, 0.8668+0.6262j],
        [0.0000+0.0000j, 0.0000+0.0000j, 0.2451+0.0000j]])
```
The reason is LAPACK uses only lower/upper triangle in computations for symmetric/Hermitian inputs.

Comment (Lezcano) [reply aten/src/ATen/native/BatchLinearAlgebra.cpp:1275]: While that is true, that relies on an undocumented behaviour.

In the documentation of [torch.cholesky](https://pytorch.org/docs/stable/generated/torch.cholesky.html) we ask for the input matrix `A` to be SPD, and this deprecation rule is correct whenever `A` s SPD.

Comment (Lezcano) [reply aten/src/ATen/native/BatchLinearAlgebra.cpp:1275]: @mruberry what do you think?

Comment (mruberry) [reply aten/src/ATen/native/BatchLinearAlgebra.cpp:1275]: Tough call because I think both perspectives are valid. Luckily, I think as long we're clear about when the deprecation rule applies, and so long as that covers all "valid" inputs, then this is OK. But we shouldn't claim equivalency on all inputs, because that implies that valid and invalid inputs would produce the same result.

Comment (Lezcano) [reply aten/src/ATen/native/BatchLinearAlgebra.cpp:1275]: Do you think that any other part of the deprecation note should be changed, or is it fine as it is?

Side note: Using this undocumented behaviour together with autograd would lead to incorrect results, as the input is upper-triangular, but the gradient is symmetric.

Comment (mruberry) [reply aten/src/ATen/native/BatchLinearAlgebra.cpp:1275]: Maybe one additional sentence at the end, just in case, like "This transform will produce equivalent results for all valid (strictly positive definite) inputs." ?

**Comment by Lezcano** [comment]
@IvanYashchuk @mruberry This should be ready.

**Review by IvanYashchuk — APPROVED** [review]

**Review by mruberry — APPROVED** [review]
Always great to have better documentation
